lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,111 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var fetchOnce = null,
+            fetchNotes,
+            renderNotes,
+            _requestsData = null,
+            storeNotesRequestData,
+            SearchRequestsData = [],
+            waitUntilAllRequestsStored;
+
+        waitUntilAllRequestsStored = function (maxTimeout) {
+            var deferred = $.Deferred(),
+                intervalId = null,
+                elapsedTimeout = 0,
+                pollInterval = 100,
+                checkRequests;
+
+            checkRequests = function() {
+               if ($('.edx-notes-wrapper').length === SearchRequestsData.length) {
```

Comment:
@bjacobel In jasmine tests `SearchRequestsData` array is already filled with previous data due to which this condition never becomes true and no search request is sent. Any suggestion to handle this in tests?